### PR TITLE
chore: cleanup LemonTable API

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We had the `display` option on the LemonTable that was only used in one place in the entire app.

## Changes

- Simplify `borderless-rows`, `stealth` and `embedded` style options to only have two
- Stealth
  - removes the background including the header
  - removes all inner borders
- Embedded
  - removes outer border and border radius
  - Does not impact background color

## How did you test this code?

Visually. All tables I was testing look the same or better

#### billing products
Screenshot 2023-11-23 at 16 16 07

#### detailed results
Screenshot 2023-11-23 at 16 16 00

#### events in notebooks
Screenshot 2023-11-23 at 16 15 53

### CODE REVIEW

File: frontend/__snapshots__/scenes-app-insights--funnel-left-to-right-breakdown--light--webkit.png
Comment (Twixes) [inline frontend/__snapshots__/scenes-app-insights--funnel-left-to-right-breakdown--light--webkit.png:1]: Let's make column group headers consistent with column headers.
BW this header text color seem very low opacity at 40%, kind of disabled. Can we use at least 50% of opacity? That's the standard for "very muted content" elsewhere in the UI

**Review by Twixes — APPROVED** [review]
Added some extra snapshots to get a better look. ;)
Looks good, though one ask from me above on the contrast side.

File: frontend/src/lib/lemon-ui/LemonTable/LemonTable.tsx [hunk 1]
```diff
@@ -52,8 +52,8 @@ export interface LemonTableProps<T extends Record<string, any>> {
     inset?: boolean
     /** An embedded table has no border around it and no background. This way it blends better into other components. */
     embedded?: boolean
-    /** Whether inner table borders should be shown. **/
-    borderedRows?: boolean
+    /** Whether a table background and inner borders should be shown. **/
```
Comment (thmsobrmlr) [inline frontend/src/lib/lemon-ui/LemonTable/LemonTable.tsx:55]: More clear to me if comments on boolean props describe what they do for the `true` case.
```suggestion
    /** Whether to hide the table background and inner borders. **/
```

**Review by thmsobrmlr — APPROVED** [review]
Makes sense to me.

**Pull request merged**